src/redis-cli.c

```diff
@@ -58,13 +58,15 @@
 #include "adlist.h"
 #include "zmalloc.h"
 #include "linenoise.h"
-#include "help.h" /* Used for backwards-compatibility with pre-7.0 servers that don't support COMMAND DOCS. */
 #include "anet.h"
 #include "ae.h"
 #include "connection.h"
 #include "cli_common.h"
 #include "mt19937-64.h"
 
+#define COMMAND_HEADER_TO_INCLUDE "cli_commands.h"
+#include "commands.c"
```

Comment:
That was my design. I love preprocessor made pasta..
I suppose your suggestion is okay as well. Or maybe I had some reason why i didn't wanna move commands.c. don't recall.

Reply:
My suggested change isn't huge if github is able to detect that a file is renamed with a few changes. Well, I've shared my thought. Now it's up to you to decide what you want to merge. :)

Reply:
@oranagra Thanks for doing this. I like it!

I too realized that it changed with the reply schema version of commands.c and I didn't quite get the time to fix it.

Reply:
Regarding the build dependencies in your last commit:

```Makefile
commands.c: $(COMMANDS_DEF_FILENAME).def

cli_commands.c: commands.def
```

The `.c` files are not generated anymore. Only the `.o` files depend on the `.c` files and the `.def` files. Shouldn't we have rules like `commands.o: commands.c commands.def` instead, or am I missing something?

> if we make the same trick here (generating `commands_with_reply_schema.def`), we'll need to keep just one commands.c and commands.o (since they do have some boilerplate code that's not generated).

One possibility is to put that boilerplate (`#define MAKE_CMD ...`) in yet another file to include. `commands_boilerplate_glue.h` or whatever...

> this way we still avoid the risk of committing `commands_with_reply_schema.def` by mistake, but maybe we have a risk of using a wrong command.o during a build.

Possibly, to avoid using the wrong `commands.o` by mistake:

```Makefile
commands.o: commands.def
commands_with_reply_schema.o: commands_with_reply_schema.def
cli_commands.o: commands.def
```